src/Http/Routing/src/EndpointMiddleware.cs

```diff
@@ -58,17 +58,34 @@ public async Task Invoke(HttpContext httpContext)
 
                 try
                 {
-                    await endpoint.RequestDelegate(httpContext);
+                    var requestTask = endpoint.RequestDelegate(httpContext);
+                    if (!requestTask.IsCompletedSuccessfully)
+                    {
+                        return AwaitRequestTask(endpoint, requestTask, _logger);
+                    }
                 }
-                finally
+                catch (Exception exception)
                 {
                     Log.ExecutedEndpoint(_logger, endpoint);
```

Comment:
What if the `requestTask` was completed synchronously, then this wouldn't get logged right?

Reply:
Good catch

Reply:
Fixed